### ISSUE TITLE
Streams in the 1.12.2 version are exceedingly rare.

### ISSUE THREAD

**Issue by StormbringerGTX** [delvr/Streams#107]
We have a 1.12.2 server and are finishing up a modpack. One of the last things we clean up is world gen. Streams was highly requested so we decided to implement that in the game. However we have noticed that the streams are extremely rare and when we do find them, they are extremely small systems.

A way to configure them or to have them outright replace vanilla rivers would be fantastic!

Keep up the great work in any case! Very interesting concept for a mod!

**Comment by xJon** [comment]
I do agree with what's written above.

**Comment by SquishyBird7** [comment]
Yes! I am also having this issue. please increase spawn rate or let us do it!

**Comment by muzikbike** [comment]
I've managed to find several easily in 1.11.2 myself. As for 1.12.2... well, i've found like 3 hollows (which are stupidly rare) and absolutely no streams.

**Comment by muzikbike** [comment]
If anyone's looking for one, the seed 102496288339226 has a stream at 1189 -215.

**Comment by delvr** [comment]
This might be related to [a workaround in version 0.4.8](https://github.com/delvr/Streams/issues/74) that fixed some issues in modpacks like Sevtech. Could you try downgrading to 0.4.7 (in a new world, not an existing one!) and see if more and/or larger streams show up then?

**Comment by SquishyBird7** [comment]
I will try this and see if it works. Also kind of unrelated, but the generation of streams with ANY cave generators I've found is nonexistent. Like with Better Caves and Worley's, no streams spawn, more like a 1 in a million chance.

**Comment by DieselpunkCyborg** [comment]
> This might be related to [a workaround in version 0.4.8](https://github.com/delvr/Streams/issues/74) that fixed some issues in modpacks like Sevtech. Could you try downgrading to 0.4.7 (in a new world, not an existing one!) and see if more and/or larger streams show up then?

I can confirm that downgrading worked. Your mod is breathtaking when its working as intended, the way little streams cut through rock is magical.

**Comment by delvr** [comment]
I've added an experimental option to Streams 0.4.9 for using the legacy Streams generation from 0.4.7. Could you guys give it a try? Here's the changelog, with instructions for the new option:
- Fixed [stone "bridges" appearing across Streams](https://github.com/delvr/Streams/issues/109#issuecomment-717107100).
- Added an option to restore the legacy Streams generation from Streams 0.4.7. Enabling legacy generation reverts a change
  in Streams 0.4.8 that prevented [infinite recursion crash in some modpacks](https://github.com/delvr/Streams/issues/74), 
  but also caused [Streams to become too rare or too small](https://github.com/delvr/Streams/issues/107).
  If you're starting a new world and not using a modpack affected by the crash, you can enable this option by either
  adding `-Dstreams.legacyGeneration=true` to your server's Java command or launcher installation's JVM arguments, 
  or setting environment variable `STREAM_LEGACY_GENERATION=true`. 
  Once enabled for a new world, that world keeps the same value even if the global setting changes later.

**Comment by TheDeviantCrafter** [comment]
**Solution: install Some like it Dry**

Stream generation is interrupted by other water sources like vanilla rivers and lakes.

Installing Some like it Dry can remove some of those. This makes them MUCH more common.

**Comment by ghost** [comment]
> I've added an experimental option to Streams 0.4.9 for using the legacy Streams generation from 0.4.7. Could you guys give it a try? Here's the changelog, with instructions for the new option:
> 
> * Fixed [stone "bridges" appearing across Streams](https://github.com/delvr/Streams/issues/109#issuecomment-717107100).
> * Added an option to restore the legacy Streams generation from Streams 0.4.7. Enabling legacy generation reverts a change
>   in Streams 0.4.8 that prevented [infinite recursion crash in some modpacks](https://github.com/delvr/Streams/issues/74),
>   but also caused [Streams to become too rare or too small](https://github.com/delvr/Streams/issues/107).
>   If you're starting a new world and not using a modpack affected by the crash, you can enable this option by either
>   adding `-Dstreams.legacyGeneration=true` to your server's Java command or launcher installation's JVM arguments,
>   or setting environment variable `STREAM_LEGACY_GENERATION=true`.
>   Once enabled for a new world, that world keeps the same value even if the global setting changes later.

How would one go about setting this up on a server?

**Comment by dudebehinddude** [comment]
the streams with the -Dstreams.legacyGeneration=true tag are definitely much more common then without it, and much longer. I didn't really test it that much, but i had to go about 2k blocks to find a stream without the tag, but with the tag spawning in there were already 3 streams in the log and they were much longer than the one i had found previously. This was done on basically a fresh vanilla world.

**Comment by ghost** [comment]
> the streams with the -Dstreams.legacyGeneration=true tag are definitely much more common then without it, and much longer. I didn't really test it that much, but i had to go about 2k blocks to find a stream without the tag, but with the tag spawning in there were already 3 streams in the log and they were much longer than the one i had found previously. This was done on basically a fresh vanilla world.

How do you set the tag to true for a singleplayer world? Or in a modpack?

**Comment by dudebehinddude** [comment]
in the minecraft launcher you edit the installation, click more options and where it says jvm arguments add `-Dstreams.legacyGeneration=true` to the end

**Comment by dudebehinddude** [comment]
if youre running a server then in the .bat file you use to run it (assuming you are running the server on a local windows machine) add it to the end of the `java -jar` part

**Comment by dudebehinddude** [comment]
adding it in the minecraft launcher only affects simgleplayer worlds btw

**Comment by ghost** [comment]
> if youre running a server then in the .bat file you use to run it (assuming you are running the server on a local windows machine) add it to the end of the `java -jar` part

What if I’m not running the server myself? I use a server hosting service, and have not seen any Java arguments.

**Comment by dudebehinddude** [comment]
what host do you use

**Comment by ghost** [comment]
GGServers

**Comment by dudebehinddude** [comment]
oh ive never used them

**Comment by dudebehinddude** [comment]
open a ticket with them and ask about it
